src/please.go

```diff
@@ -309,6 +309,14 @@ var opts struct {
 				Targets []core.BuildLabel `position-arg-name:"targets" description:"Additional targets to load rules from"`
 			} `positional-args:"true"`
 		} `command:"rules" description:"Prints built-in rules to stdout as JSON"`
+		Changes struct {
+			Before          string `short:"b" long:"before" description:"Revision to check out for the state before"`
+			After           string `short:"a" long:"after" required:"true" description:"Revision to check out for the state after"`
```

Comment:
Why do we have both `before` and `after`? Naively I would expect this to work as follows:

`plz query changed` with no arguments compares the current working directory state to the last commit (i.e. HEAD, i.e. a noop when directly on a git commit).

`plz query changed --since [reflike]` compare the current working directory state to the given reflike.

So for CI I would expect to run something like `plz query changed --since @{upstream} (or origin/master I guess)`. And on merge I'd either run everything or just `plz query changed --since HEAD^` (given our workflow) - I don't think I'd ever want to do anything else.

Either way I think before and after are slightly confusing terms here.

Reply:
That is fair.

Either it's a copy-and-paste job from the old flow, or I was trying to answer the question of how I know what the current revision is - i.e. the logic for `plz query changes --since origin/master` looks like:
```
git checkout origin/master
plz do stuff
git checkout ????
plz do more stuff
```
I do agree it's a nicer interface - maybe another pseudo-flag to work out the current revision? I am trying not to insist on git here although obviously it's somewhat biased towards it - but I think that's okay as long as it's possible to choose something else via a sufficient supply of flags.